lib/iris/tests/unit/experimental/ugrid/test_load_meshes.py

```diff
@@ -0,0 +1,218 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Unit tests for the :func:`iris.experimental.ugrid.load_meshes` function.
+
+"""
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests  # isort:skip
+
+from pathlib import Path
+from shutil import rmtree
+from subprocess import check_call
+import tempfile
+from unittest import mock
+from uuid import uuid4
+
+from iris.experimental.ugrid import PARSE_UGRID_ON_LOAD, load_meshes, logger
+
+
+def setUpModule():
+    global TMP_DIR
+    TMP_DIR = Path(tempfile.mkdtemp())
+
+
+def tearDownModule():
+    if TMP_DIR is not None:
+        rmtree(TMP_DIR)
+
+
+def cdl_to_nc(cdl):
+    cdl_path = TMP_DIR / "tst.cdl"
+    nc_path = TMP_DIR / f"{uuid4()}.nc"
+    # Write CDL string into a temporary CDL file.
+    with open(cdl_path, "w") as f_out:
+        f_out.write(cdl)
+    # Use ncgen to convert this into an actual (temporary) netCDF file.
+    command = "ncgen -o {} {}".format(nc_path, cdl_path)
+    check_call(command, shell=True)
+    return str(nc_path)
+
+
+class TestsBasic(tests.IrisTest):
+    def setUp(self):
+        self.ref_cdl = """
+            netcdf mesh_test {
+                dimensions:
+                    node = 3 ;
+                    face = 1 ;
+                    vertex = 3 ;
+                    levels = 2 ;
+                variables:
+                    int mesh ;
+                        mesh:cf_role = "mesh_topology" ;
+                        mesh:topology_dimension = 2 ;
+                        mesh:node_coordinates = "node_x node_y" ;
+                        mesh:face_coordinates = "face_x face_y" ;
+                        mesh:face_node_connectivity = "face_nodes" ;
+                    float node_x(node) ;
+                        node_x:standard_name = "longitude" ;
+                    float node_y(node) ;
+                        node_y:standard_name = "latitude" ;
+                    float face_x(face) ;
+                        face_x:standard_name = "longitude" ;
+                    float face_y(face) ;
```

Comment:
I suspect that face coordinates are redundant for what you are testing here.
If so, I would omit them for minimalism + brevity

Reply:
https://github.com/SciTools/iris/pull/4262#discussion_r692873231

Reply:
d8809da8baec16684f98f281574ace3b8bc7438d